lib/ui/settings/profile/share_profile_qr_scan_screen.dart

```diff
@@ -180,14 +199,28 @@ class _ShareProfileQrScanScreenState extends ConsumerState<ShareProfileQrScanScr
         return;
       case AppLifecycleState.resumed:
         _subscription = _controller.barcodes.listen(_handleBarcode);
-        unawaited(_controller.start());
+        unawaited(_safeStartCamera());
       case AppLifecycleState.inactive:
         unawaited(_subscription?.cancel());
         _subscription = null;
         unawaited(_controller.stop());
     }
   }
 
+  Future<void> _safeStartCamera() async {
```

Comment:
This logic is duplicated beween 2 widgets, I think it would be better to have it in one single place